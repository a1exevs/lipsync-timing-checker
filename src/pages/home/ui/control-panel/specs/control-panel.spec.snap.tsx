import { render } from '@testing-library/react';
import React from 'react';

import ControlPanel from 'src/pages/home/ui/control-panel/control-panel';

describe('ControlPanel', () => {
  const baseProps = {
    isPlaying: false,
    isAudioPlayButtonEnabled: true,
    onPlayPause: jest.fn(),
    isCaretLocked: false,
    isPinCaretButtonEnabled: true,
    onLockCaretButtonClick: jest.fn(),
    playDuringDrag: false,
    isPlayDuringDragButtonEnabled: true,
    setPlayDuringDrag: jest.fn(),
  };

  it('matches snapshot (default)', () => {
    const { container } = render(<ControlPanel {...baseProps} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot (playing, caret locked, playDuringDrag)', () => {
    const { container } = render(<ControlPanel {...baseProps} isPlaying isCaretLocked playDuringDrag />);
    expect(container).toMatchSnapshot();
  });
});
