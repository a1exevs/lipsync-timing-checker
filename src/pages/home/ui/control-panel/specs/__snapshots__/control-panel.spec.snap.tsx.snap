// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ControlPanel matches snapshot (default) 1`] = `
<div>
  <section
    class="flex items-center gap-6 my-4 rounded-md bg-gray-800 p-4 shadow"
  >
    <button
      class="rounded-full transition p-2 w-10 h-10 bg-blue-600 hover:bg-blue-700 text-white disabled:cursor-not-allowed disabled:opacity-50"
      title="Play"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-play"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="6 3 20 12 6 21 6 3"
        />
      </svg>
    </button>
    <div
      class="flex items-center gap-2"
    >
      <button
        class="rounded-full transition p-2 w-10 h-10 bg-gray-100 hover:bg-gray-200 text-gray-800 border border-gray-300 disabled:cursor-not-allowed disabled:opacity-50"
        title="Lock caret position"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-pin"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 17v5"
          />
          <path
            d="M9 10.76a2 2 0 0 1-1.11 1.79l-1.78.9A2 2 0 0 0 5 15.24V16a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-.76a2 2 0 0 0-1.11-1.79l-1.78-.9A2 2 0 0 1 15 10.76V7a1 1 0 0 1 1-1 2 2 0 0 0 0-4H8a2 2 0 0 0 0 4 1 1 0 0 1 1 1z"
          />
        </svg>
      </button>
      <button
        class="rounded-full transition p-2 w-10 h-10 bg-gray-100 hover:bg-gray-200 text-gray-800 border border-gray-300 disabled:cursor-not-allowed disabled:opacity-50"
        title="Enable audio while dragging"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-ear"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 8.5a6.5 6.5 0 1 1 13 0c0 6-6 6-6 10a3.5 3.5 0 1 1-7 0"
          />
          <path
            d="M15 8.5a2.5 2.5 0 0 0-5 0v1a2 2 0 1 1 0 4"
          />
        </svg>
      </button>
    </div>
  </section>
</div>
`;

exports[`ControlPanel matches snapshot (playing, caret locked, playDuringDrag) 1`] = `
<div>
  <section
    class="flex items-center gap-6 my-4 rounded-md bg-gray-800 p-4 shadow"
  >
    <button
      class="rounded-full transition p-2 w-10 h-10 bg-blue-600 hover:bg-blue-700 text-white disabled:cursor-not-allowed disabled:opacity-50"
      title="Pause"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-pause"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="16"
          rx="1"
          width="4"
          x="14"
          y="4"
        />
        <rect
          height="16"
          rx="1"
          width="4"
          x="6"
          y="4"
        />
      </svg>
    </button>
    <div
      class="flex items-center gap-2"
    >
      <button
        class="rounded-full transition p-2 w-10 h-10 bg-blue-600 hover:bg-blue-700 text-white disabled:cursor-not-allowed disabled:opacity-50"
        title="Unlock caret position"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-pin"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 17v5"
          />
          <path
            d="M9 10.76a2 2 0 0 1-1.11 1.79l-1.78.9A2 2 0 0 0 5 15.24V16a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-.76a2 2 0 0 0-1.11-1.79l-1.78-.9A2 2 0 0 1 15 10.76V7a1 1 0 0 1 1-1 2 2 0 0 0 0-4H8a2 2 0 0 0 0 4 1 1 0 0 1 1 1z"
          />
        </svg>
      </button>
      <button
        class="rounded-full transition p-2 w-10 h-10 bg-blue-600 hover:bg-blue-700 text-white disabled:cursor-not-allowed disabled:opacity-50"
        title="Disable audio while dragging"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-ear"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 8.5a6.5 6.5 0 1 1 13 0c0 6-6 6-6 10a3.5 3.5 0 1 1-7 0"
          />
          <path
            d="M15 8.5a2.5 2.5 0 0 0-5 0v1a2 2 0 1 1 0 4"
          />
        </svg>
      </button>
    </div>
  </section>
</div>
`;
